fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneAnalyzerFactory.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * LuceneAnalyzerFactory.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.lucene;
+
+import com.apple.foundationdb.record.metadata.Index;
+import com.apple.foundationdb.record.provider.foundationdb.indexes.TextIndexMaintainer;
+import org.apache.lucene.analysis.Analyzer;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Each implementation of {@link Analyzer} should have its own implementation
+ * of this factory interface to provide instances of the analyzer to a
+ * {@link LuceneAnalyzerRegistry}. The registry will populate a mapping of names to
+ * analyzers using the methods of this interface.
+ *
+ * @see TextIndexMaintainer
```

Comment:
Why this and not the `Lucene` index maintainer?

Reply:
Good catch! Because I just copied the Javadoc from ```TextTokenizerFactory``` and edited it, but missed this line. Fixed it now.